pkg/image/export/main.go

```diff
@@ -265,13 +269,69 @@ func getK3sUpgradeImages(rancherVersion string, k3sData map[string]interface{})
 			}
 		}
 
-		version = strings.Replace(version, "+", "-", -1)
-		k3sImages = append(k3sImages, fmt.Sprintf(imageFormat, version))
+		compatibleReleases = append(compatibleReleases, version)
+	}
+
+	imageFormat := "rancher/%s"
+	for _, release := range compatibleReleases {
+		// registries don't allow +, so image names will have these substituted
+		upgradeImage := fmt.Sprintf(imageFormat, "k3s-upgrade:"+strings.Replace(release, "+", "-", -1))
+		k3sImagesMap[upgradeImage] = true
+
+		images, err := downloadK3sSupportingImages(release)
+		if err != nil {
+			logrus.Infof("could not find supporting images for k3s release [%s]: %v", release, err)
+			continue
+		}
+
+		supportingImages := strings.Split(images, "\n")
+		if supportingImages[len(supportingImages)-1] == "" {
+			supportingImages = supportingImages[:len(supportingImages)-1]
+		}
+
+		for _, imageName := range supportingImages {
+			// ignore original registry and repo name
+			imageParts := strings.Split(imageName, "/")
+			aliasedImageName := fmt.Sprintf(imageFormat, imageParts[1]+"-"+imageParts[2])
```

Comment:
dont manipulate image names here. just remove registry if it is `docker.io` (remove prefix)